public/pages/DefineDetector/containers/DefineDetector.tsx

```diff
@@ -218,6 +262,61 @@ export const DefineDetector = (props: DefineDetectorProps) => {
     }
   };
 
+  const handleDataSourceChange = ([event]) => {
+    const dataSourceId = event?.id;
+    if (!dataSourceId) {
+      getNotifications().toasts.addDanger(
+        prettifyErrorMessage('Unable to set data source.')
+      );
```

Comment:
not so sure... MDS team provided this usage in their component guideline doc

Reply:
The callback function will either return `[]` (error state), `[{id: '', label: 'Local cluster']` (if `Local cluster` is enabled) or `[{id: <dataSourceId>, label: <dataSourceTitle>}]`. Thus, `event` can be undefined and it's expected that Plugins handle this case as an invalid/error state. 

One example of `[]` being returned is when you pass in an `activeOption` and you specify a `dataSourceFilter` that filters out the datasource. Since the `dataSourceFilter` function removes the `activeOption` as a valid choice, the component will return `[]`.